### PR TITLE
feat: implement deep linking for message signing

### PR DESCRIPTION
**Pull request by alfonsobries**
## Summary

Closes https://app.clickup.com/t/2p89rtg

## Checklist

-   [x] My changes look good in both light AND dark mode
-   [x] The change is not hardcoded to a single network, but has multi-asset in mind
-   [x] I checked my changes for obvious issues, debug statements and commented code
-   [ ] Documentation _(if necessary)_
-   [x] Tests _(if necessary)_
-   [x] Ready to be merged

### CODE REVIEW

**Comment by ItsANameToo** [comment]
@alfonsobries I changed the base to `feat/message-sign-deeplinking`, can you open any additional PRs against that branch so we combine it all in there before merging it into develop upon completion?

File: src/app/hooks/use-search-parameters-validation.ts [hunk 1]
```diff
@@ -38,9 +46,52 @@ export const useSearchParametersValidation = () => {
...
+					return `${generatePath(ProfilePaths.SignMessageWallet, {
+						profileId: profile.id(),
+						walletId: wallet.id(),
+					})}?${parameters.toString()}`;
+				}
+
+				return `${generatePath(ProfilePaths.SignMessage, {
+					profileId: profile.id(),
+				})}?${parameters.toString()}`;
+			},
```
Comment (dated) [inline src/app/hooks/use-search-parameters-validation.ts:87]:
```suggestion
			path: ({ profile, parameters }: PathProperties) =>
				`${generatePath(ProfilePaths.SignMessage, {
					profileId: profile.id(),
				})}?${parameters.toString()}`,
```

When deeplinking we always use the paths without wallet context.

**Review by samharperpittam — CHANGES_REQUESTED** [review]
- Wallet with 0 balance can't sign a message, found by @dated 
image


- ~When trying to sign with a Ledger wallet, the following error can be found in console
`SignMessage.tsx:110 Uncaught (in promise) TypeError: Cannot read properties of undefined (reading 'coinId')`~ fixed

**Review by samharperpittam — CHANGES_REQUESTED** [review]
~Show error if message content exceeds the allowed limit~fixed in [f2bd0a6](https://github.com/ArdentHQ/arkvault/pull/138/commits/f2bd0a65c9b88955050504e42408084041d005d6)

image

**Review by samharperpittam — APPROVED** [review]

**Review by goga-m — APPROVED** [review]

**Review by samharperpittam — APPROVED** [review]

**Pull request merged**